src/rust/ide/view/graph-editor/src/component/visualization/container.rs

```diff
@@ -596,28 +578,27 @@ impl Container {
             }));
 
             eval fullscreen_position.value ((p) model.fullscreen_view.set_position(*p));
-            eval model.frp.preprocessor    ((code) inputs.preprocessor_select.emit(code));
         }
 
 
         // ===  Visualisation chooser frp bindings ===
         frp::extend! { network
-            eval action_bar.visualisation_selection([model,registry,scene,action_bar](visualization_path) {
+            eval action_bar.visualisation_selection(
+            [model,registry,action_bar,frp](visualization_path) {
```

Comment:
I've refactored this to be a bit nicer and void the memory leak. Unfortunately, filter doesn't work here, as the `definition` needs to be wrapped in an `Option` because it does not itself implement `Default.`